src/functions-and-arrays.js

```diff
@@ -1,41 +1,113 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num2 > num1) {
+    return num2;
+  } else {
+    return num1;
+  }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(words) {
+  let longestWord = '';
+  if (words.length === 0) {
+    return null;
+  }
+  words.forEach((word) => {
+    if (word.length > longestWord.length) {
+      longestWord = word;
+    } 
+  });
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  let sum = 0;
+  if ((numbers.length === 0)) {
+    return 0;
+  }
+  numbers.forEach((number) => {
+    sum += number;
+  });
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function sum(mixedArr) {
+  let sum = 0;
+  mixedArr.forEach((number) => {
+    if (typeof number != 'string' && typeof number != 'boolean' && typeof number != 'number'){
+      throw new Error("Unsupported data type sir or ma'am")
+    } else if ((mixedArr.length === 0)) {
+      return 0;
+    } else if (typeof number == 'string') {
```

Comment:
After all the null- and error-checks I would start with a new if-statement instead of else-if, that makes the code more readable